src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -131,6 +134,35 @@ Frame::Frame(
             impl->m_frame_height,
             impl->m_tile_width,
             impl->m_tile_height));
+
+    // Copy and add the aovs.
+    const AOVFactoryRegistrar aov_registrar;
+    for (size_t i = 0, e = aovs.size(); i < e; ++i)
+    {
+        const AOV* original_aov = aovs.get_by_index(i);
+
+        const size_t aov_index = aov_images().get_index(original_aov->get_name());
+        if (aov_index == size_t(~0) && aov_images().size() < MaxAOVCount)
+        {
+            const IAOVFactory* aov_factory = aov_registrar.lookup(original_aov->get_model());
+            assert(aov_factory);
+
+            auto_release_ptr<AOV> aov = aov_factory->create(original_aov->get_parameters());
+
+            aov_images().append(
+                aov->get_name(),
+                4, // aov->get_channel_count(),
```

Comment:
Why is 4 hardcoded here?

Reply:
It comes from the original code. I added the comment to not forget about this.
I didn't check yet if there's any issue passing the correct number of channels here.